runtime/lua/vim/lsp/actions.lua

```diff
@@ -0,0 +1,343 @@
+--[=[
+
+actions.lua
+
+Define common actions that can be taken on different LSP structures.
+
+Module structure is:
+
+  actions.<lsp_structure_name>.<action_name>
+
+  Where:
+    lsp_structure_name:
+      Name of the structure as defined in the LSP specification.
+
+      For example: Location, LocationLink, Hover, etc.
+
+      All structures are assumed to be (wherever possible) to be either a singular or list of those items.
+        The LSP spec often has something return either `Location` or `Location[]`.
+        All callbacks here should be able to handle either of them.
+
+
+    action_name:
+      Name of what the user can expect to happen when this function is called.
+
+
+  Example:
+    actions.Location.jump_first will jump to the singular location passed or the first location in a list of locations.
```

Comment:
Change working to `Location` or similar.

Should probably add some documentation / links for these somewhere sensible.